scripts/attack.py

```diff
@@ -0,0 +1,179 @@
+        version=attack_config["model_version"],
+        id_postfix=trainer_config["wandb"]["id_postfix"],
+    )
+    model_A.eval()
+    model_B.eval()
+
+    # Prepare dmpair
+    dmpair_kwargs = opts2dmpairArgs(
+        opt=dataset_config["experiment_groups"][experiment_group][dataset_index],
+        seed=dataset_config["seeds"][seed_index],
+    )
+    dmpair = DMPair(**dmpair_kwargs)
+
+    # Get base images from test datasets
+    images_A, labels_A, images_B, labels_B = dmpair.sample_from_test_pairs(
+        num_attack_images
+    )
+
+    # Generate adversial images
+    # 2 models * 2 distributions * 2 attacks = 8
+    fga_images = generate_over_combinations(
+        model_A=model_A,
+        model_B=model_B,
+        images_A=images_A,
+        labels_A=labels_A,
+        images_B=images_B,
+        labels_B=labels_B,
+        attack_fn_name=attack_names[0],
+        epsilons=epsilons,
+    )
+    boundary_images = generate_over_combinations(
+        model_A=model_A,
+        model_B=model_B,
+        images_A=images_A,
+        labels_A=labels_A,
+        images_B=images_B,
+        labels_B=labels_B,
+        attack_fn_name=attack_names[1],
+        epsilons=epsilons,
+        steps=500,
+    )
+
+    # Transfer attack over model*dist combinations, comptue succes
+    # Names: AB_to_B imples attack trained on model A with images from distribution B,
+    # transferred to model B
+    transfer_metrics_AA_to_B = compute_transfer_attack(
+        model=model_B,
+        images=images_A,
+        labels=labels_A,
+        advs_images=[fga_images["model_A_dist_A"], boundary_images["model_A_dist_A"]],
+        attack_names=attack_names,
+    )
+    transfer_metrics_AB_to_B = compute_transfer_attack(
+        model=model_B,
+        images=images_B,
+        labels=labels_B,
+        advs_images=[fga_images["model_A_dist_B"], boundary_images["model_A_dist_B"]],
+        attack_names=attack_names,
+    )
+    transfer_metrics_BA_to_A = compute_transfer_attack(
+        model=model_A,
+        images=images_A,
+        labels=labels_A,
+        advs_images=[fga_images["model_B_dist_A"], boundary_images["model_B_dist_A"]],
+        attack_names=attack_names,
+    )
+    transfer_metrics_BB_to_A = compute_transfer_attack(
+        model=model_A,
+        images=images_B,
+        labels=labels_B,
+        advs_images=[fga_images["model_B_dist_B"], boundary_images["model_B_dist_B"]],
+        attack_names=attack_names,
+    )
+
+    # Output
+    A_to_B_metrics = {
+        "dist_A": transfer_metrics_AA_to_B,
+        "dist_B": transfer_metrics_AB_to_B,
+    }
+    B_to_A_metrics = {
+        "dist_A": transfer_metrics_BA_to_A,
+        "dist_B": transfer_metrics_BB_to_A,
+    }
+    run_A.log({"A_to_B_metrics": A_to_B_metrics}, commit=True)
+    run_B.log({"B_to_A_metrics": B_to_A_metrics}, commit=True)
+
+    # Close once finished
+    run_A.finish()
+    run_B.finish()
+
+    return A_to_B_metrics, B_to_A_metrics
```

Comment:
don't return here as we don't intend to call `main` to get a value back

Reply:
done